### PR TITLE
4_GET_api_articles

### PR DESCRIPTION
**Pull request by AlexandraB123**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -37,3 +37,35 @@ describe("/api/topics", () => {
...
+describe("/api/articles", () => {
+  test("GET:200. Sends an array of articles to the client", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        expect(res.body.articles).toEqual(expect.any(Array));
+        expect(res.body.articles.length).toBeGreaterThan(0);
+        res.body.articles.forEach((article) => {
+          article.comment_count = Number(article.comment_count)
```
Comment (MKD01) [inline __tests__/api.test.js]: This logic shouldn't be done in the your test.

File: models/topics.model.js [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
-    .then((result) => result.rows)
-}
\ No newline at end of file
+  return db.query("SELECT * FROM topics;").then((result) => result.rows);
+};
+
+exports.fetchArticles = () => {
+    const queryString = `SELECT users.username AS author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, COUNT(comments.comment_id) as comment_count 
+    FROM articles
+    LEFT JOIN users ON articles.author = users.username
```
Comment (MKD01) [inline models/topics.model.js]: Do you need to join to the articles table to access the username? or do you already have access to it?

File: models/topics.model.js [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+    LEFT JOIN comments ON articles.article_id = comments.article_id
+    GROUP BY users.username, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes
```
Comment (MKD01) [inline models/topics.model.js]: Do you need to be grouping by anything other than articles.article_id here?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Please do make these changes and send through another PR to reflect this.

File: models/topics.model.js (hunk 3 above)
Comment (AlexandraB123) [reply models/topics.model.js]: Grouping updated to article_id only

File: models/topics.model.js (hunk 2 above)
Comment (AlexandraB123) [reply models/topics.model.js]: Join to users removed. articles.author column used instead

File: __tests__/api.test.js (hunk 1 above)
Comment (AlexandraB123) [reply __tests__/api.test.js]: Logic removed. SQL query string now casts as integer

**Pull request merged**